### DISCUSSION TITLE
我们制作了新的 VCD 波形渲染器，大家可以畅所欲言讲出你们想要的设计

### DISCUSSION THREAD

**Discussion by LSTM-Kirigaya** [Digital-EDA/Digital-IDE#57] [Ideas]
目前做了一个比较简单的雏形：

## 0.3.3 版本

<img width="1617" alt="a2521a6e103ee9e91a9fcf124e09fe05" src="https://github.com/Digital-EDA/Digital-IDE/assets/59416203/ba771309-676a-4d63-b90b-f1fe25f94d28">

未来在功能和渲染上都会持续更新。但是目前我们希望大家可以结合实际开发的需求和经验提出你们想要的功能。我们后期会采纳。

## 0.4.0 版本

![82f0b6b7715ec955d4dfab9494701371](https://github.com/user-attachments/assets/243fca6b-65b1-4972-b82b-e67fe1eb6412)

**Comment by DuBirdFly** [comment]
1. 需要上下拖动移动波形功能
2. 需要把波形合并成 group 功能
3. 需要插件自动保存上一次添加的信号, 不然每次不小心关了 *.vcd 的标签页再打开还挺麻烦的
4. zoom fit 功能
5. next edge 查下一个边沿功能


还有可能是 SV 还是 UVM 的原因, vcd 对变量的位宽放在后面
这里 gtkwave 和 modelsim 都能把变量当作一个变量, dide 暂时还不行
![image](https://github.com/Digital-EDA/Digital-IDE/assets/98462076/25b50c65-e4cc-4c00-8cd8-bd3c4e7d6177)
![image](https://github.com/Digital-EDA/Digital-IDE/assets/98462076/b09778f9-8a8b-4af8-a8cc-05d853bd8453)
![image](https://github.com/Digital-EDA/Digital-IDE/assets/98462076/23995200-d732-4b51-a020-50551889e0d6)

**Reply by DuBirdFly** [reply]
[vsim.vcd.zip](https://github.com/Digital-EDA/Digital-IDE/files/15369974/vsim.vcd.zip)

**Reply by LSTM-Kirigaya** [reply]
感谢您的反馈！想问一下 “需要上下拖动移动波形功能“ 指的是什么。目前我们的 viewer 在波形数量很多时，是可以上下滚动鼠标来查看的。

波形合并成 group 功能听着很不错，你的预期是类似于 chrome 浏览器上的标签页分组的功能吗？

![image](https://github.com/Digital-EDA/Digital-IDE/assets/59416203/7cdd079b-a20f-41ec-8892-4dbf2a9f97e0)

**Reply by DuBirdFly** [reply]
1. 合并 group 确实像标签页分组
![PixPin_group](https://github.com/Digital-EDA/Digital-IDE/assets/98462076/873853fa-2bed-44fc-add9-4ef7fa31a75a)

2. 移动波形 (一般是拖动信号名, WaveTrace插件似乎是用的 CTRL+↑/↓)
![PixPin_move](https://github.com/Digital-EDA/Digital-IDE/assets/98462076/36c845b4-8247-4c68-8142-2f08049b95b0)

**Reply by LSTM-Kirigaya** [reply]
您好，我们已经实现了除了自动保存外你想要的所有功能，新版本 vcd 渲染器将在 2024 中秋节版本的更新中出现，下面是功能预览：


https://github.com/user-attachments/assets/4e656072-4085-49c6-aa2b-79db57098a6c

https://github.com/user-attachments/assets/6ad1abe3-a894-42fc-89ab-fc1745851e0e

**Comment by Tianli-Wang** [comment]
1. vcd查看器能不能直接一整个模块都放进去，现在好像只能每个信号手动点，信号多了有点麻烦。比如右键模块名可以选择整个模块查看波形，或者多选信号可以一起查看。
2. 波形查看保存当前排版，下次打开还能看到调整好的波形，不要清空重来。
<img width="499" alt="Snipaste_2024-08-21_09-52-54" src="https://github.com/user-attachments/assets/b13f14dd-ac97-4c6a-b879-a81bcf2d843e">

**Reply by LSTM-Kirigaya** [reply]
您好，除了自动保存之外，连续选中波形的功能在新版本中已经实现，通过按住 shift 加上点击鼠标左击就可以连续选中一堆波形，这个操作同时适用于左侧的 sidebar 和右侧的 波形选择视窗中：


https://github.com/user-attachments/assets/b460daaf-098c-47f3-b372-3aeda070713c
